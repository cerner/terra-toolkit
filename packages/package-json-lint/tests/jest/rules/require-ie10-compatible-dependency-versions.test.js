const requireIe10CompatibleDependencyVersions = require('../../../src/rules/require-ie10-compatible-dependency-versions');

describe('require-ie10-compatible-dependency-versions', () => {
  it('succeeds when there are no dependencies', () => {
    const results = requireIe10CompatibleDependencyVersions.create({
      ruleConfig: {
        severity: {
          severityType: 'warn',
        },
      },
      projectType: 'module',
    }).dependencies({});
    expect(results).toMatchSnapshot();
  });

  it('succeeds when there are dependencies not in the list to check', () => {
    const results = requireIe10CompatibleDependencyVersions.create({
      ruleConfig: {
        severity: {
          severityType: 'warn',
        },
      },
      projectType: 'module',
    }).dependencies({
      a: '^1.0.0',
    });
    expect(results).toMatchSnapshot();
  });

  it('fails as a warning when versions do not meet the IE10 compatible version', () => {
    let results;
    requireIe10CompatibleDependencyVersions.create({
      ruleConfig: {
        severity: {
          severityType: 'warn',
        },
      },
      projectType: 'module',
      report: (issues) => {
        results = issues;
      },
    }).dependencies({
      redux: '^5.0.0',
    });
    expect(results).toMatchSnapshot();
  });

  it('fails when versions do not meet the IE10 compatible version', () => {
    let results;
    requireIe10CompatibleDependencyVersions.create({
      ruleConfig: {
        severity: {
          severityType: 'error',
        },
      },
      projectType: 'devModule',
      report: (issues) => {
        results = issues;
      },
    }).dependencies({
      uuid: '7.x.x',
    });
    expect(results).toMatchSnapshot();
  });

  it('passes when all versions meet the IE10 compatible version', () => {
    const results = requireIe10CompatibleDependencyVersions.create({
      ruleConfig: {
        severity: {
          severityType: 'warn',
        },
      },
      projectType: 'module',
    }).dependencies({
      redux: '<3.0.0',
      axios: '<0.18.1',
      uuid: '<6.0.0',
    });
    expect(results).toMatchSnapshot();
  });

  it('succeeds when versions do not meet the IE10 compatible version but package is passed in the allowList', () => {
    const results = requireIe10CompatibleDependencyVersions.create({
      ruleConfig: {
        severity: {
          severityType: 'warn',
          allowList: ['redux'],
        },
      },
      projectType: 'module',
    }).dependencies({
      redux: '^4.0.0',
    });
    expect(results).toMatchSnapshot();
  });
});
