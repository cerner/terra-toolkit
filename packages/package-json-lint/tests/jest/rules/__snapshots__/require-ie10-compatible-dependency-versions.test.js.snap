// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`require-ie10-compatible-dependency-versions fails as a warning when versions do not meet the IE10 compatible version 1`] = `
Object {
  "lintId": "require-ie10-compatible-dependency-versions",
  "lintMessage": "The dependencies for this project do not satisfy the requirement for require-ie10-compatible-dependency-versions:
  redux@^5.0.0 does not satisfy IE10 compatibility for require-ie10-compatible-dependency-versions. **Note** Use redux@3.7.2 or lower instead.",
  "projectType": "module",
  "severity": Object {
    "severityType": "warn",
  },
}
`;

exports[`require-ie10-compatible-dependency-versions fails when versions do not meet the IE10 compatible version 1`] = `
Object {
  "lintId": "require-ie10-compatible-dependency-versions",
  "lintMessage": "The dependencies for this project do not satisfy the requirement for require-ie10-compatible-dependency-versions:
  uuid@7.x.x does not satisfy IE10 compatibility for require-ie10-compatible-dependency-versions. **Note** Use uuid@3.4.0 or lower instead.",
  "projectType": "devModule",
  "severity": Object {
    "severityType": "error",
  },
}
`;

exports[`require-ie10-compatible-dependency-versions passes when all versions meet the IE10 compatible version 1`] = `undefined`;

exports[`require-ie10-compatible-dependency-versions succeeds when there are dependencies not in the list to check 1`] = `undefined`;

exports[`require-ie10-compatible-dependency-versions succeeds when there are no dependencies 1`] = `undefined`;

exports[`require-ie10-compatible-dependency-versions succeeds when versions do not meet the IE10 compatible version but package is passed in the allowList 1`] = `undefined`;
